File: Server/PenetrationProjectAPI/src/main/java/com/ssafy/cafe/model/service/CouponService.java
package com.ssafy.cafe.model.service;

import java.util.List;

import com.ssafy.cafe.model.dto.Coupon;

public interface CouponService {

	List<Coupon> getCouponByUser(String userId);
	
	public void addCoupon(Coupon coupon);
}

File: SmartStoreUI/app/src/main/java/com/ssafy/smartstore/service/CouponService.kt
package com.ssafy.smartstore.service

import com.ssafy.smartstore.dto.CouponDto
import com.ssafy.smartstore.dto.OrderDetailDto
import com.ssafy.smartstore.dto.OrderDto
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface CouponService {
    @GET("rest/coupon/{userId}")
    fun getCouponByUser(@Path("userId") userId: String): Call<List<CouponDto>>

    @POST("rest/coupon")
    fun addCoupon(@Body body: CouponDto): Call<Int>
}